src/read.c

```diff
@@ -799,6 +799,16 @@ static avifResult avifMetaFindOrCreateItem(avifMeta * meta, uint32_t itemID, avi
         }
     }
 
+    uint32_t prevID = 0;
+    // ISO/IEC 23008-12, First edition, 2017-12, Section 9.3.1:
+    //   Each ItemPropertyAssociation box shall be ordered by increasing item_ID, and there shall
+    //   be at most one association box for each item_ID, in any ItemPropertyAssociation box.
+    for (uint32_t i = 0; i < meta->items.count; ++i) {
+        AVIF_CHECKERR(meta->items.item[i]->id > prevID, AVIF_RESULT_BMFF_PARSE_FAILED);
```

Comment:
It would be good to add a diagnostic message. (We won't be able to use the `AVIF_CHECKERR` macro.)

Reply:
Done (i had to change all callers)